backend/backend.go

```diff
@@ -91,6 +91,11 @@ type Backend interface {
 	// [api.ErrInstanceNotFound] is returned if the specified orchestration instance doesn't exist.
 	// [api.ErrNotCompleted] is returned if the specified orchestration instance is still running.
 	PurgeOrchestrationState(context.Context, api.InstanceID) error
+
+	// CleanupOrchestration clean up all records for the specified orchestration instance in the entire task hub.
+	//
+	// [api.ErrInstanceNotFound] is returned if the specified orchestration instance doesn't exist.
+	CleanupOrchestration(context.Context, api.InstanceID) error
```

Comment:
Can we accomplish this without introducing a new backend API? For example, could we just use `PurgeOrchestrationState`? Alternatively, could we just bake this logic into `CreateOrchestrationInstance`? I want to be careful about adding new backend APIs unless there's a strong reason to do so.

Reply:
> Alternatively, could we just bake this logic into CreateOrchestrationInstance? I want to be careful about adding new backend APIs unless there's a strong reason to do so.

The reason I didn't put this logic into `CreateOrchestrationInstance` is that I want to keep minimum updates from the implementation side. But definitely I can move the logic there. 
`PurgeOrchestrationState` only clean up the `History` and `Instances` tables but didn't clean up `NewEvents` and  `NewTasks` tables.

Reply:
Ah, purge should probably clean up `NewEvents` and `NewTasks` too. That's just a missing part of the implementation.